### PR TITLE
chore(frontend): Avoid redundant v2 API call in PipelineDetails.

### PR DESCRIPTION
**Pull request by jlyaoyuli**
Part of #9731 
Previously, we update the URL with `pipelineVersionId` obtained in getSelectedVersion(), and it will cause to re-render the page, which brings out several redundant API calls. The PR is removing the URL update logic to avoid those calls. Also, the getPipelineVersion() in _getTemplateString() is not necessary either, so it's removed as well.

### CODE REVIEW

File: frontend/src/pages/PipelineDetails.tsx [hunk 1]
```diff
@@ -296,21 +296,18 @@ class PipelineDetails extends Page<{}, PipelineDetailsState> {
...
         if (listVersionsResponse.pipeline_versions) {
-          latesetVersion = listVersionsResponse.pipeline_versions[0];
-          // Append version id to URL for create run (new run switcher call getPipelineVersion)
-          this.props.history.replace({
-            pathname: `/pipelines/details/${pipelineId}/version/${latesetVersion.pipeline_version_id}`,
-          });
-          return latesetVersion;
+          selectedVersion = listVersionsResponse.pipeline_versions[0];
+        } else {
+          return;
```
Comment (zijianjoy) [inline frontend/src/pages/PipelineDetails.tsx:301]: Can we `return undefined`  in this case?

File: frontend/src/pages/PipelineDetails.tsx [hunk 2]
```diff
@@ -296,21 +296,18 @@ class PipelineDetails extends Page<{}, PipelineDetailsState> {
...
         }
-        return undefined;
       } catch (err) {
         this.setStateSafe({ graphIsLoading: false });
         await this.showPageError('Cannot retrieve pipeline version list.', err);
         logger.error('Cannot retrieve pipeline version list.', err);
         return;
```
Comment (zijianjoy) [inline frontend/src/pages/PipelineDetails.tsx:307]: Can we `return undefined` in this case?

File: frontend/src/pages/PipelineDetails.tsx [hunk 3]
```diff
@@ -296,21 +296,18 @@ class PipelineDetails extends Page<{}, PipelineDetailsState> {
         );
 
         if (listVersionsResponse.pipeline_versions) {
```
Comment (zijianjoy) [inline frontend/src/pages/PipelineDetails.tsx:298]: We should probably also check that `len(listVersionsResponse.pipeline_versions) > 0`

File: frontend/src/pages/PipelineDetails.tsx [hunk 4]
```diff
@@ -523,10 +520,7 @@ class PipelineDetails extends Page<{}, PipelineDetailsState> {
           logger.error('Cannot retrieve pipeline versions.', err);
           return;
         }
-        templateString = await this._getTemplateString(
-          pipelineId,
-          v2SelectedVersion ? v2SelectedVersion.pipeline_version_id! : v1SelectedVersion?.id!,
-        );
```
Comment (zijianjoy) [inline frontend/src/pages/PipelineDetails.tsx:529]: Can we remove `v1SelectedVersion` as well? Just curious if we can rely on v2 API as much as we can.

File: frontend/src/pages/PipelineDetails.test.tsx [hunk 5]
```diff
@@ -535,12 +535,12 @@ describe('PipelineDetails', () => {
 
     await waitFor(() => {
       // one for selected Version, another for template string
```
Comment (zijianjoy) [inline frontend/src/pages/PipelineDetails.test.tsx:537]: This comment needs update as well.

File: frontend/src/pages/PipelineDetails.tsx [hunk 6]
```diff
@@ -296,21 +296,18 @@ class PipelineDetails extends Page<{}, PipelineDetailsState> {
         );
 
         if (listVersionsResponse.pipeline_versions) {
-          latesetVersion = listVersionsResponse.pipeline_versions[0];
-          // Append version id to URL for create run (new run switcher call getPipelineVersion)
-          this.props.history.replace({
-            pathname: `/pipelines/details/${pipelineId}/version/${latesetVersion.pipeline_version_id}`,
-          });
-          return latesetVersion;
+          selectedVersion = listVersionsResponse.pipeline_versions[0];
```
Comment (zijianjoy) [inline frontend/src/pages/PipelineDetails.tsx:299]: Can you add a test to verify that: ` If there are more than 1 pipeline version, the first version is being used`.  The way to verify that the first version is being used is to check the API call to getPipelineVersion is using the first version ID in this request.

File: frontend/src/pages/PipelineDetails.tsx (hunk 4 above)
Comment (jlyaoyuli) [reply frontend/src/pages/PipelineDetails.tsx:529]: Plan to do it in next PR if it's workable : )

File: frontend/src/pages/PipelineDetails.test.tsx (hunk 5 above)
Comment (zijianjoy) [inline frontend/src/pages/PipelineDetails.test.tsx]: The comment here needs to be updated.

**Comment by zijianjoy** [comment]
/lgtm

Only one minor comment, thank you Joe!

**Pull request merged**